storage-proofs/core/src/crypto/pedersen.rs

```diff
@@ -405,8 +405,8 @@ mod tests {
             let hashed = pedersen_md_no_padding(&flat);
 
             let mut hasher = Hasher::new(&x[0]).unwrap();
-            for k in 1..5 {
-                hasher.update(&x[k]).unwrap();
+            for val in x.iter().take(5).skip(1) {
```

Comment:
Just a personal nitpick (no need to change it). I would write it as `x.iter().skip(1).take(5)`. I needed a while to parse this one, I read the current version as "take the first five elements, ohhh wait, skip the first one".

Reply:
Agreed.  Highlighting that you are correct is the fact that the `5` is misleading because you translated it to `x.iter().skip(1).take(5)` which iis different, we actually only want 4 items. I'll use `x.iter().skip(1).take(4)` for this and I'll use this order for this sort of combinator in future, thanks for the lesson.